src/Couchbase.Lite.Shared/API/Sync/Replicator.cs

```diff
@@ -472,7 +472,6 @@ private void OnDocEnded(List<ReplicatedDocument> replications, bool pushing)
                 var logDocID = new SecureLogString(docID, LogMessageSensitivity.PotentiallyInsecure);
                 if (!pushing && error.domain == C4ErrorDomain.LiteCoreDomain &&
                     error.code == (int)C4ErrorCode.Conflict) {
-                    replications.Remove(replication);
```

Comment:
This could easily result in two document ended events for a conflict.  Once with "conflict" and one with no error after the conflict resolution is done.  This needs to stay here.